### PR TITLE
Nordic: Port privacy to softdevice v4.

### PR DESCRIPTION
**Pull request by pan-**
### Description

Many things have changed; the identity list isn't shared anymore with the whitelist and resolution is handled by the stack itself.

### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

### CODE REVIEW

File: features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xPalSecurityManager.cpp [hunk 1]
```diff
@@ -999,6 +975,14 @@ bool nRF5xSecurityManager::sm_handler(const ble_evt_t *evt)
                             address_type = advertising_peer_address_type_t::RANDOM_ADDRESS;
                         }
 
+                        printf("peer address distributed: ");
```
Comment (paul-szczepanek-arm) [inline features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xPalSecurityManager.cpp]: ?

File: features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xGap.cpp [hunk 2]
```diff
@@ -1322,32 +1374,16 @@ void nRF5xGap::on_connection(Gap::Handle_t handle, const ble_gap_evt_connected_t
...
     using BLEProtocol::AddressType;
 
-    AddressType_t peer_addr_type;
-    const uint8_t* peer_address = evt.peer_addr.addr;
-
     if (_privacy_enabled &&
-        evt.peer_addr.addr_type == BLE_GAP_ADDR_TYPE_RANDOM_PRIVATE_RESOLVABLE &&
-        _central_privacy_configuration.resolution_strategy != CentralPrivacyConfiguration_t::DO_NOT_RESOLVE
+        evt.peer_addr.addr_id_peer == 0 &&
+        _central_privacy_configuration.resolution_strategy == CentralPrivacyConfiguration_t::RESOLVE_AND_FILTER
```
Comment (donatieng) [inline features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xGap.cpp]: I assume we cannot implement RESOLVE_AND_FORWARD with the Softdevice?

File: features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xPalSecurityManager.cpp [hunk 3]
```diff
@@ -200,50 +208,18 @@ ble_error_t nRF5xSecurityManager::remove_device_from_resolving_list(
 
 ble_error_t nRF5xSecurityManager::clear_resolving_list()
 {
+    printf("clear resolving list\r\n");
```
Comment (donatieng) [inline features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xPalSecurityManager.cpp]: Should be removed :)

File: features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xGap.cpp (hunk 2 above)
Comment (pan-) [reply features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xGap.cpp]: Resolve and forward **is** implemented but a bit differently that earlier: Previously resolution was happening when an address was received; now it is made by the stack. 
As a result, `DO_NOT_RESOLVE`, disable resolution while other levels enable it. 
When a packet is received we only filter when `RESOLVE_AND_FILTER` is set as resolution already happened.

Comment (donatieng) [reply features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF52/source/nRF5xGap.cpp]: Cool, thanks for the clarification

**Comment by pan-** [comment]
@donatieng @paul-szczepanek-arm  I think I've addressed your commits.

**Pull request merged**